numba/tests/test_parallel_backend.py

```diff
@@ -954,6 +955,91 @@ def forker():
+            from numba.pycc.platform import Toolchain, _external_compiler_ok
+            from numba import njit, prange, threading_layer
+
+            if not _external_compiler_ok:
+                raise AssertionError('External compilers are not found.')
+
+            try:
+                with TemporaryDirectory() as tmpdir:
+                    with NamedTemporaryFile(dir=tmpdir) as tmpfile:
+                        src = \"\"\"
+                        #define TBB_PREVIEW_WAITING_FOR_WORKERS 1
+                        #include <tbb/tbb.h>
+                        static tbb::task_scheduler_handle tsh;
+                        extern "C"
+                        {
+                        void launch(void)
+                        {
+                            tsh = tbb::task_scheduler_handle::get();
+                        }
+                        }
+                        \"\"\"
+                        cxxfile = f"{tmpfile.name}.cxx"
+                        with open(cxxfile, 'wt') as f:
+                            f.write(src)
+                        tc = Toolchain()
+                        object_files = tc.compile_objects([cxxfile,],
+                                                           output_dir=tmpdir)
+                        dso_name = f"{tmpfile.name}.so"
+                        tc.link_shared(dso_name, object_files,
+                                       libraries=['tbb',],
+                                       export_symbols=['launch'])
+                        # Load into the process, it doesn't matter whether the
+                        # DSO exists on disk once it's loaded in.
+                        DLL = ctypes.CDLL(dso_name)
+            except Exception as e:
+                # Something is broken in compilation, could be one of many
+                # things including, but not limited to: missing tbb headers,
+                # incorrect permissions, compilers that don't work for the above
+                print(e)
+                print('BROKEN_COMPILERS')
+                sys.exit(0)
+
+            # Do the test
+            DLL.launch()
+
+            @njit(parallel=True)
+            def foo(n):
+                acc = 0
+                for i in prange(n):
+                    acc += i
+                return acc
+
+            foo(1)
+            assert threading_layer() == 'tbb'
+            pid = os.fork()
+            if pid != 0:
+                # parent prints success
+                print("SUCCESS")
+            sys.exit(0)
+            """.replace('BROKEN_COMPILERS', BROKEN_COMPILERS)
+
+        cmdline = [sys.executable, '-c', runme]
+        env = os.environ.copy()
+        env['NUMBA_THREADING_LAYER'] = 'tbb'
+        out, err = self.run_cmd(cmdline, env=env)
+
+        if BROKEN_COMPILERS in out:
+            self.skipTest("Compilation of DSO failed. Check output for details")
+        else:
+            self.assertIn("SUCCESS", out)
```

Comment:
Shouldn't success be tested by via exit code == 0?

Reply:
I've attempted to explain what's going with respect to this in in the above. There's essentially some aliasing of "cannot run this test due to compiler problem" and "test succeeded", the "test failed" case ends up as an assertion.

If you want to try out the `BROKEN_COMPILERS` branch, compiler Numba with `tbb` and `tbb-devel` present, then remove `tbb-devel` (so there's no `tbb.h` header) and run this test, it should skip.